### PR TITLE
Explicitly specify proper branch and change to support depends-on

### PR DESCRIPTION
**Pull request by jianghuaw**

### CODE REVIEW

File: fuel_test/localrc.default [hunk 1]
```diff
@@ -10,6 +9,10 @@ export FUEL_TEST_LOG_DIR=${FUEL_TEST_LOG_DIR:-"fuel_test_logs"}
 export FM_SNAPSHOT="fuel_ci"
 export ENV_NAME="TestEnv"
 
+export BRANCH=${ZUUL_BRANCH:-"master"}
+export MERGE_REPO_URL=${MERGE_REPO_URL:-"https://review.openstack.org/p"}
```
Comment (BobBall) [inline fuel_test/localrc.default]: The two defaults of 'master' and 'review.openstack.org/p' don't make sense together

File: fuel_test/check_version.sh [hunk 2]
```diff
@@ -7,10 +7,10 @@
 [ $DEBUG == "on" ] && set -x
 
 if [[ ! -d "/tmp/fuel-plugin-xenserver" ]]; then
-	git clone https://review.openstack.org/openstack/fuel-plugin-xenserver /tmp/fuel-plugin-xenserver
+	git clone https://review.openstack.org/openstack/fuel-plugin-xenserver -b ${BRANCH} /tmp/fuel-plugin-xenserver
```
Comment (BobBall) [inline fuel_test/check_version.sh]: I'm confused - why is/was this cloning from review.openstack.org? Surely it should be git.openstack.org?

File: fuel_test/deploy_env.sh [hunk 3]
```diff
@@ -34,10 +34,10 @@ yum install python-pip git createrepo dpkg-devel dpkg-dev rpm rpm-build -y
 cd /root/
 
 if [[ ! -d "fuel-plugin-xenserver" ]]; then
-	git clone https://review.openstack.org/openstack/fuel-plugin-xenserver
+	git clone https://review.openstack.org/openstack/fuel-plugin-xenserver -b '${BRANCH}'
```
Comment (BobBall) [inline fuel_test/deploy_env.sh]: why review.openstack.org?

File: fuel_test/check_version.sh [hunk 4]
```diff
@@ -7,10 +7,10 @@
...
 fi
 cd /tmp/fuel-plugin-xenserver
-git fetch https://review.openstack.org/openstack/fuel-plugin-xenserver "$FUEL_PLUGIN_REFSPEC"
```
Comment (BobBall) [inline fuel_test/check_version.sh]: I think this fetch is the only place we needed review.openstack.org in the original (pre-depends-on) case?

File: fuel_test/check_version.sh (hunk 2 above)
Comment (huazhihao) [inline fuel_test/check_version.sh]: Yes, git.openstack.org is better. But review.openstack.org is functionally equal. Previously I copy the command from gerrit (the checkout link). We can have it replaced here too,

**Pull request merged**